File: Selenium/Activities/Activity14.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity14 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println("Title: " + driver.getTitle());
        List<WebElement> rows=driver.findElements(By.xpath("//table[@id='sortableTable']/thead/tr"));
        System.out.println("Number of rows: "+rows.size());
        List<WebElement> cols=driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("Number of cols: "+cols.size());
        WebElement cellvalue=driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println(cellvalue.getText());
        driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th")).click();
        WebElement cellvalue_sort=driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println("After sort:"+cellvalue_sort.getText());
        List<WebElement> footer = driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        System.out.println("Cell values of the footer: ");
        for(WebElement foot : footer) {
            System.out.println(foot.getText());
        }
        driver.close();

    }

    }

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    driver.get("https://v1.training-support.net/selenium/tables")
    print("Title: " + driver.title)

    rows=driver.find_elements(By.XPATH,"//table[@id='sortableTable']/tbody/tr")
    print("Number of rows: ",len(rows))
    cols=driver.find_elements(By.XPATH,"//table[@id='sortableTable']/thead/tr/th")
    print("Number of cols: ",len(cols))
    cellvalue=driver.find_element(By.XPATH,"//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print(cellvalue.text)
    driver.find_element(By.XPATH,"//table[@id='sortableTable']/thead/tr/th").click()
    cellvalue_sort=driver.find_element(By.XPATH,"//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print("After sort:"+cellvalue_sort.text)
    footer = driver.find_elements(By.XPATH,"//table[@id='sortableTable']/tfoot/tr/th")
    print("Cell values of the footer: ")
    for foot in footer:
        print(foot.text)

    driver.close()



